docker_images/mbed-os-env/Dockerfile

```diff
@@ -0,0 +1,103 @@
+# ------------------------------------------------------------------------------
+# Pull base image
+FROM ubuntu:20.04
+
+# ------------------------------------------------------------------------------
+# Arguments
+ARG WORKDIR=/root
+
+# ------------------------------------------------------------------------------
+# Install tools via apt
+ENV DEBIAN_FRONTEND=noninteractive
+RUN set -x \
+    && apt -y update \
+    && apt -y install \
+    git \
+    wget \
+    python3 \
+    python3-dev \
+    python3-setuptools \
+    python3-usb \
+    python3-pip \
+    software-properties-common \
+    build-essential \
+    astyle \
+    mercurial \
+    ninja-build \
+    libssl-dev \
+    cargo \
+    flex \
+    bison \
+    && apt clean && rm -rf /var/lib/apt/lists \
+    && update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 \
+    && : # last line
+
+RUN set -x \
+    && pip3 install -U cmake
+
+WORKDIR /tmp/
+
+RUN set -x \
+    && git clone --depth=1 --single-branch --branch Release_1_8_14 https://github.com/doxygen/doxygen \
+    && cd doxygen \
+    && mkdir build \
+    && cd build \
+    && cmake -G "Unix Makefiles" .. \
+    && make \
+    && make install
+
+# Set up mbed environment
+WORKDIR /tmp/
+COPY requirements.txt .
+RUN set -x \
+    && pip3 install -r requirements.txt  \
```

Comment:
Consider also installing the CMake-specific requirements, via `tools/cmake/requirements.txt`. I suspect they are a subset of what's in `requirements.txt` at the moment, but that could change.

Reply:
ok.